### PR TITLE
odoo fixes

### PR DESCRIPTION
**Pull request by zimbatm**
## Description of changes

Banging Odoo 17 into shape.

- **odoo: add missing lxml-html-clean dependency**
- **odoo: add missing geoip2 dependency**
- **nixos/odoo: use env vars to configure Odoo**
- **nixos/odoo: add autoInit option**
- **nixos/odoo: set data_dir explicitly**

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by zimbatm** [comment]
A follow-up to https://github.com/NixOS/nixpkgs/pull/327641

File: nixos/modules/services/finance/odoo.nix [hunk 1]
```diff
@@ -113,7 +115,9 @@ in
           "HOME=%S/odoo"
           "ODOO_RC=${cfgFile}"
         ];
-      };
+      } // (lib.optionalAttrs cfg.autoInit {
+        ExecStartPre = "${cfg.package}/bin/odoo --init=INIT --database=odoo --db_user=odoo --stop-after-init";
```
Comment (siriobalmelli) [inline nixos/modules/services/finance/odoo.nix]: I'm pessimistic that all init steps are idempotent, such that running this before each start won't mess up a database with runtime data.

Might I suggest something like:

```bash
INITIALIZED="${cfg.data_dir}/.odoo.initialized"
if [ ! -e "$INITIALIZED" ]; then
  "${cfg.package}/bin/odoo --init=INIT --database=odoo --db_user=odoo --stop-after-init"
  touch "$INITIALIZED"
fi
```

**Review by siriobalmelli — APPROVED** [review]

**Comment by siriobalmelli** [comment]
I also recommend removing the following from `nixos/tests/odoo.nix`:

```nix
      # odoo does not automatically initialize its database,
      # even if passing what _should_ be the equivalent of these options:
      #  settings = {
      #    options = {
      #      database = "odoo";
      #      init = "base";
      #    };
      #  };
      systemd.services.odoo.preStart = ''
        HOME=$STATE_DIRECTORY ${package}/bin/odoo -d odoo -i base --stop-after-init --without-demo all
      '';
```

**Comment by siriobalmelli** [comment]
Result of `nixpkgs-review pr 327729` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>odoo</li>
    <li>odoo.dist</li>
  </ul>
</details>

**Comment by zimbatm** [comment]
Thanks, that's good feedback. I will stop force-pushing if you want to add your changes on top. Otherwise I'll get back to it tomorrow.

File: nixos/modules/services/finance/odoo.nix (hunk 1 above)
Comment (zimbatm) [reply nixos/modules/services/finance/odoo.nix]: :+1: It seems to work, but better safe than sorry.

Comment (mkg20001) [reply nixos/modules/services/finance/odoo.nix]: I assume, if plugins are changed init does not need to be re-run? Then :+1:

**Comment by siriobalmelli** [comment]
I'm on it - will revert in 60 min or so

**Review by siriobalmelli — COMMENTED** [review]
LGTM, proposed changes as per my previous message

**Comment by zimbatm** [comment]
thanks, pushed your patches and waiting on ofBorg

**Pull request merged**